### DISCUSSION TITLE
Is it possible to implement masking using CustomDepth and SceneDepth?

### DISCUSSION THREAD

**Discussion by helirang** [RyanSweeney987/distance-field-outline-docs#1] [Q&A] [answered]
I recently bought and tested this asset. So far, I think this is the best unreal jfa outline asset I've seen. Thank you for selling it. I need a masking feature like the question, but I don't know how to compare customDepth and SceneDepth in the ush file. Is it possible to add masking feature to the asset in that way?

Turning on customdepth for the objects I want to be hidden works as I want, but it's too difficult to maintain.

![a1](https://github.com/user-attachments/assets/c28d4eee-5846-4be8-a047-d842ecbfa330)
![a2](https://github.com/user-attachments/assets/669ed5d5-b906-4468-9b9b-76cc0a5e296a)

**Comment by RyanSweeney987** [comment]
I'll have a look at it, it should be doable but I can't remember how things are set up to talk about comparing CustomDepth and SceneDepth in the ush

I'll add it as a feature either way

**Comment by RyanSweeney987** [comment, marked as answer]
Hi, just messaging to let you know that scene depth occlusion has been added in for the update 1.3 - sorry it took so long
I didn't include custom depth as it already occludes stencils, unless I'm misunderstanding what you mean. Either way, including won't take long, especially not half a year XD

Here are some examples of just the outline being occluded
![Screenshot 2025-06-20 020116](https://github.com/user-attachments/assets/a43f0fff-4c3f-45b3-9d80-16550a9cf623)
![Screenshot 2025-06-20 020125](https://github.com/user-attachments/assets/19a0b025-b5d1-4179-b6e0-58adb64136dc)

And like your images here's with the stencil itself being occluded
![Screenshot 2025-06-20 020134](https://github.com/user-attachments/assets/f44bcf89-3471-4ede-b5ef-a1a0fb58ac48)
![Screenshot 2025-06-20 020140](https://github.com/user-attachments/assets/d1029a1c-8f7f-4d28-a6bf-817a8987087c)

You also get access to both the stencil and occluded pixel data in the output render target - I'll update appropriate material nodes
